tests/integration/Invitation/InvitationServiceTest.php

```diff
@@ -0,0 +1,133 @@
+    {
+        $now = (new \DateTime())->getTimestamp();
+        $invitation = $this->invitationService->createInvitation(
+            new InvitationCreateStruct(
+                'thisisjustatest@ibexa.co',
+                'admin',
+            )
+        );
+
+        self::assertEquals('thisisjustatest@ibexa.co', $invitation->getEmail());
+        self::assertEquals('admin', $invitation->getSiteAccessIdentifier());
+        self::assertNotNull($invitation->getHash());
+        self::assertFalse($invitation->isUsed());
+        self::assertEqualsWithDelta($now, $invitation->createdAt()->getTimestamp(), 10);
+    }
+
+    public function testCreateInvitationForExistingUser(): void
+    {
+        $this->expectExceptionObject(new UserAlreadyExistsException());
+        $this->invitationService->createInvitation(
+            new InvitationCreateStruct(
+                'admin@link.invalid',
+                'admin',
+            )
+        );
+    }
+
+    public function testCreateExistingInvitation(): void
+    {
+        $this->expectExceptionObject(new InvitationAlreadyExistsException());
+        $this->invitationService->createInvitation(
+            new InvitationCreateStruct(
+                'thisisjustatest@ibexa.co',
+                'admin',
+            )
+        );
+    }
+
+    public function testFindInvitations(): void
+    {
+        $invitations = $this->invitationService->findInvitations();
+        self::assertCount(1, $invitations);
+    }
+
+    public function testFindInvitationsWithUserGroupFilter(): void
+    {
+        $editorsGroup = $this->userService->loadUserGroupByRemoteId('3c160cca19fb135f83bd02d911f04db2');
+
+        $this->invitationService->createInvitation(
+            new InvitationCreateStruct(
+                'editor@ibexa.co',
+                'admin',
+                $editorsGroup
+            )
+        );
+
+        $invitations = $this->invitationService->findInvitations();
+        self::assertCount(2, $invitations);
+
+        $editorInvitations = $this->invitationService->findInvitations(
+            new InvitationFilter(null, $editorsGroup)
+        );
+
+        self::assertCount(1, $editorInvitations);
+    }
+
+    public function testMarkInvitationAsUsed(): void
+    {
+        $invitation = $this->invitationService->createInvitation(
+            new InvitationCreateStruct(
+                'used@ibexa.co',
+                'admin',
+            )
+        );
+
+        self::assertFalse($invitation->isUsed());
+
+        $this->invitationService->markAsUsed($invitation);
+
+        self::assertTrue(
+            $this->invitationService->getInvitation($invitation->getHash())->isUsed()
+        );
+    }
+
+    public function testRefreshInvitation(): void
+    {
+        $invitation = $this->invitationService->createInvitation(
+            new InvitationCreateStruct(
+                'refresh@ibexa.co',
+                'admin',
+            )
+        );
+        sleep(5);
```

Comment:
Can we reduce this value, 5 seconds on the entire test suite seems to excessive. Isn't 1s enough to have a difference?

Reply:
Seems like a candidate for a `ClockMock`.